plugins/cache/cache_types.ml

```diff
@@ -0,0 +1,44 @@
+open Core_kernel
+open Regular.Std
+
+type entry = {
```

Comment:
Given that our cache entry deteriorated to just a path (with probably unnecessary size, as we can get it from stat), I now have a question: why do we need index at all? 

Like the natural step forward would be to use the digest as the name of the file and rely on stat and readdir to perform cleanup. We also, don't need a global file lock anymore, so that different instances of bap could work in parallel without any congestion.